core/templates/dev/head/services/UserServiceSpec.ts

```diff
@@ -17,18 +17,79 @@
  */
 
 describe('User Service', function() {
-  var UserService, $httpBackend, UrlInterpolationService;
-
+  var UserService, $httpBackend, UrlInterpolationService,
+    UserInfoObjectFactory;
   beforeEach(angular.mock.module('oppia'));
   beforeEach(angular.mock.inject(function($injector) {
     UserService = $injector.get('UserService');
     UrlInterpolationService = $injector.get(
       'UrlInterpolationService');
+    UserInfoObjectFactory = $injector.get(
+      'UserInfoObjectFactory');
     $httpBackend = $injector.get('$httpBackend');
   }));
 
+
+  it('should be return userInfo data', function() {
+    // creating a test user for checking profile picture of user.
+    var sampleUserInfoBackendObject = {
+      is_moderator: false,
+      is_admin: false,
+      is_super_admin: false,
+      is_topic_manager: false,
+      can_create_collections: true,
+      preferred_site_language_code: null,
+      username: 'tester',
+      user_is_logged_in: true
+    };
+
+    $httpBackend.expect('GET', '/userinfohandler').respond(200,
+      sampleUserInfoBackendObject);
+
+    var sampleUserInfo = UserInfoObjectFactory.createFromBackendDict(
+      sampleUserInfoBackendObject);
+
+    UserService.getUserInfoAsync().then(function(userInfo) {
+      expect(userInfo.is_admin).toBe(sampleUserInfo.is_admin);
+      expect(userInfo.is_super_admin).toBe(sampleUserInfo.is_super_admin);
+      expect(userInfo.is_moderator).toBe(sampleUserInfo.is_moderator);
+      expect(userInfo.is_topic_manager).toBe(sampleUserInfo.is_topic_manager);
+      expect(userInfo.user_is_logged_in).toBe(
+        sampleUserInfo.user_is_logged_in);
+      expect(userInfo.can_create_collections).toBe(
+        sampleUserInfo.can_create_collections);
+      expect(userInfo.username).toBe(sampleUserInfo.username);
+      expect(userInfo.preferred_site_language_code).toBe(
+        sampleUserInfo.preferred_site_language_code);
+    });
+    $httpBackend.flush();
+
+    // second call to /userinfohandler.
+    $httpBackend.expect('GET', '/userinfohandler').respond(200);
```

Comment:
This seems wrong. You shouldn't be using try/catch in tests.

How many times do you expect /userinfohandler to be called? If you're expecting it to be called twice and it's only called once then we need to investigate why it's only called once.

Reply:
@seanlip I am expecting the /userinfohandler to be called only once. That's why I made second call to /userinforhandler and except it to give an error. I am not sure how to do that other than try/catch. That's why I use try/catch. because I know `$httpBackend.flush()` will give an error.
I tried this `expect($httpbackend.flush()).toThrowError()`.

Reply:
Look into verifyNoOutstandingRequest, maybe that will help you. I think we might even do it automatically already, not sure.

I'm not sure why you're making the call itself here -- you're just testing your own test. If you want to make sure that the Oppia code doesn't call that handler more than once, you need to verify that using a "spy" or a call counter (or equivalent, like verifyNoOutstandingRequest).